src/main/java/org/homeschoolpebt/app/interceptor/FlowSwitchInterceptor.java

```diff
@@ -0,0 +1,55 @@
+package org.homeschoolpebt.app.interceptor;
+
+import formflow.library.data.SubmissionRepositoryService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+@Component
+@Slf4j
+public class FlowSwitchInterceptor implements HandlerInterceptor {
+  public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
+  private final SubmissionRepositoryService submissionRepositoryService;
+
+  public FlowSwitchInterceptor(SubmissionRepositoryService submissionRepositoryService) {
+    this.submissionRepositoryService = submissionRepositoryService;
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+    try {
+      var parsedUrl = new AntPathMatcher().extractUriTemplateVariables(PATH_FORMAT, request.getRequestURI());
+      String urlFlow = parsedUrl.get("flow");
+      var submissionIdFromSession = (UUID) request.getSession().getAttribute("id");
```

Comment:
I'm going to update mine - we should use `getSession(false)` everywhere we don't intend to make a new session.